src/main/scala/io/scalac/tezos/translator/routes/MessageRoutes.scala

```diff
@@ -21,10 +22,15 @@ class MessageRoutes(
   import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
   import io.circe.generic.auto._
 
+  implicit val ec = actorSystem.dispatcher
   override def routes: Route =
     (path ("message") & pathEndOrSingleSlash & withReCaptchaVerify(log, reCaptchaConfig)(actorSystem)
       & withSendMessageValidation & post) { sendEmail =>
-      val operationPerformed = service.addNewEmail2Send(SendEmail.fromSendEmailRoutesDto(sendEmail, adminEmail))
+      val operationPerformed = for {
+        sendEmail <-  Future.fromTry(SendEmail.fromSendEmailRoutesDto(sendEmail, adminEmail))
+        y         <-  service.addNewEmail2Send(sendEmail)
```

Comment:
`y` ? please change it to smth more descriptive. If its a unit you can even do it in this way:
```
  _         <-  service.addNewEmail2Send(sendEmail)
} yield ()
```